task/power/src/main.rs

```diff
@@ -643,6 +657,66 @@ impl idl::InOrderPowerImpl for ServerImpl {
 
         Ok(device.pmbus_read(op)?)
     }
+
+    fn bmr491_event_log_read(
+        &mut self,
+        _msg: &userlib::RecvMessage,
+        index: u8,
+    ) -> Result<Bmr491Event, idol_runtime::RequestError<ResponseCode>> {
+        if index >= 48 {
+            return Err(ResponseCode::BadArg.into());
+        }
+
+        let dev = self.bmr491()?;
+        dev.write(&[
+            pmbus::commands::bmr491::CommandCode::MFR_EVENT_INDEX as u8,
+            index,
+        ])?;
+
+        let out = dev.read_reg(
+            pmbus::commands::bmr491::CommandCode::MFR_READ_EVENT as u8,
+        )?;
+
+        Ok(out)
+    }
+
+    fn bmr491_max_fault_event_index(
+        &mut self,
+        _msg: &userlib::RecvMessage,
+    ) -> Result<u8, idol_runtime::RequestError<ResponseCode>> {
+        let dev = self.bmr491()?;
+
+        // 255 is a special value, setting MFR_EVENT_INDEX to the index of the
+        // newest record in the fault section of the event recorder.
+        dev.write(&[
+            pmbus::commands::bmr491::CommandCode::MFR_EVENT_INDEX as u8,
+            255,
+        ])?;
+
+        let out = dev.read_reg(
+            pmbus::commands::bmr491::CommandCode::MFR_EVENT_INDEX as u8,
+        )?;
+        Ok(out)
+    }
+
+    fn bmr491_max_lifecycle_event_index(
+        &mut self,
+        _msg: &userlib::RecvMessage,
+    ) -> Result<u8, idol_runtime::RequestError<ResponseCode>> {
+        let dev = self.bmr491()?;
+
+        // 255 is a special value, setting MFR_EVENT_INDEX to the index of the
```

Comment:
I assume this should be

```suggestion
        // 254 is a special value, setting MFR_EVENT_INDEX to the index of the
```